flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -279,21 +270,55 @@ protected void initialize() throws ResourceManagerException {
 			throw new ResourceManagerException("Unable to configure the artifact server with TaskManager artifacts.", e);
 		}
 
-		// begin scheduling
-		connectionMonitor.tell(new ConnectionMonitor.Start(), selfActor);
-		schedulerDriver.start();
-
 		LOG.info("Mesos resource manager initialized.");
 	}
 
+	@Override
+	protected CompletableFuture<Void> prepareLeadershipAsync() {
+		return getRpcService().execute(() -> recoverWorkers())
+			.thenAcceptAsync((tasksFromPreviousAttempts) -> {
+				try {
+					recoverWorkers(tasksFromPreviousAttempts);
+				} catch (Exception e) {
+					throw new CompletionException(new ResourceManagerException(e));
+				}
+
+				// begin scheduling
+				connectionMonitor.tell(new ConnectionMonitor.Start(), selfActor);
+				schedulerDriver.start();
+
+				LOG.info("Mesos resource manager started.");
+			}, getMainThreadExecutor());
+	}
+
+	@Override
+	protected void clearState() {
+		schedulerDriver.stop(true);
```

Comment:
From my experiments it seems that it is not possible to stop and re-start the `SchedulerDriver`. I'd suggest re-creating the supporting actors and the `SchedulerDriver`.